.github/workflows/release.yml

```diff
@@ -2,42 +2,14 @@ name: Build and Release
 
 on:
   workflow_run:
-    workflows: ["Test"]
-    branches:
-      - main
+    workflows: [Test]
     types:
       - completed
 
 jobs:
-  version:
-    runs-on: ubuntu-latest
-    if: github.event.workflow_run.conclusion == 'success' && github.event_name == 'push'
-    steps:
-      - name: Checkout main branch
-        uses: actions/checkout@v3
-        with:
-          ref: 'main'
-          ssh-key: ${{ secrets.SSH_KEY }}
-      - name: Set version
-        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
-      - name: Update version
-        run: python scripts/update_version.py ${{ env.RELEASE_VERSION }}
-      - name: Auto-commit changes
-        uses: stefanzweifel/git-auto-commit-action@49620cd3ed21ee620a48530e81dba0d139c9cb80
-        with:
-          commit_message: Bump version to ${{ env.RELEASE_VERSION }}
-          create_branch: true
-          branch: release-${{ env.RELEASE_VERSION }}
-      - name: Merge changes to main
-        run: |
-          git checkout "main"
-          git merge "release-${{ env.RELEASE_VERSION }}" --no-edit
-          git push
-
-
   dist:
     runs-on: ubuntu-latest
-    needs: [version]
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
```

Comment:
why are we dropping `github.event_name == 'push'`?

Reply:
I believe it's redundant because only a pushed tag pointing to the current commit will return `success`.
Given the commit validation in `Test`, we can probably remove `workflow_dispatch` from `Test` as well. It will only ever fail